addons/mail/static/tests/tours/discuss_channel_call_public_tour.js

```diff
@@ -0,0 +1,31 @@
+import { registry } from "@web/core/registry";
+
+registry.category("web_tour.tours").add("discuss_channel_call_public_tour.js", {
+    test: true,
+    steps: () => [
+        {
+            content: "The call does not start on the welcome page",
+            trigger: ".o-mail-WelcomePage",
+            async run() {
+                await new Promise((r) => setTimeout(r, 250));
+                if (odoo.__WOWL_DEBUG__.root.env.services["discuss.rtc"]?.selfSession) {
+                    console.error("The call should not have started.");
+                }
+            },
```

Comment:
`250` can be too slow and too fast, slow due to increasing runbot time considerably for this test, and slow because this can be prone to runbot errors, notably if RPC takes more than 250ms.

A solution could be to check `!selfSession` but also no ongoing `join_call` rpc, the latter can have some state presence in rtc service

Reply:
good idea, there is already the `rtcService.state.hasPendingRequest` for that, I will add it